### PR TITLE
fix: correct return types

### PR DESCRIPTION
**Pull request by Tuditi**
# Description of change

Fixed return types and renamed transfer -> transaction to align with domain language.

## Links to any relevant issues

Be sure to reference any related issues by adding `fixes issue #`.

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Bug fix (a non-breaking change which fixes an issue)
- Documentation Fix

## How the change has been tested

Ran test, clippy and formatter

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: bindings/nodejs/examples/4b-send-transfer.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ async function run() {
         console.log('Account:', account);
 
         //TODO: test this once outputs can be conveniently built
-        // const response = await account.sendTransfer([
+        // const response = await account.SendTransaction([
```
Comment (Thoralf-M) [inline bindings/nodejs/examples/4b-send-transfer.js]:
```suggestion
        // const response = await account.sendTransaction([
```

File: documentation/docs/libraries/python/api_reference.md [hunk 2]
```diff
@@ -442,11 +442,11 @@ Returns the [AccountSynchronizer](#accountsynchronizer) to setup the process to
 
 ### transfer(transfer_obj): [WalletBlock](#walletblock)
 
-Transfer tokens.
+Transaction tokens.
```
Comment (thibault-martinez) [inline documentation/docs/libraries/python/api_reference.md]: Wrong ?

File: documentation/docs/libraries/nodejs/api_reference.md [hunk 3]
```diff
@@ -156,9 +156,9 @@ Synchronize all stored accounts with the Tangle.
 
 Returns a promise resolving to an array of [SyncedAccount](#syncedaccount).
 
-### internalTransfer(fromAccount, toAccount, amount)
+### internalTransaction(fromAccount, toAccount, amount)
 
-Transfers an amount from one sub-account to another.
+Transactions an amount from one sub-account to another.
```
Comment (thibault-martinez) [inline documentation/docs/libraries/nodejs/api_reference.md]: Wrong ?

File: src/account/operations/transaction/mod.rs [hunk 4]
```diff
@@ -50,16 +48,17 @@ impl AccountHandle {
     /// Send a transaction, if sending a block fails, the function will return None for the block_id, but the wallet
     /// will retry sending the transaction during syncing.
     /// ```ignore
-    /// let outputs = vec![TransferOutput {
-    ///     address: "atoi1qpszqzadsym6wpppd6z037dvlejmjuke7s24hm95s9fg9vpua7vluehe53e".to_string(),
+    /// let outputs = vec![TransactionOutput {
+    ///     address: "// transaction or transaction?
```
Comment (Thoralf-M) [inline src/account/operations/transaction/mod.rs]: No transaction or transaction?

File: bindings/nodejs/CHANGELOG.md [hunk 5]
```diff
@@ -230,7 +230,7 @@
   - [9855cfa](https://github.com/iotaledger/wallet.rs/commit/9855cfa4ce7296d04d1c647c7f6ca1722784eb33) refactor(manager): `sync_accounts` gap_limit and address_index options ([#346](https://github.com/iotaledger/wallet.rs/pull/346)) on 2021-02-24
 - Move `transfer`, `retry`, `reattach`, `promote` APIs to the account object.
   - [8b808c8](https://github.com/iotaledger/wallet.rs/commit/8b808c80bbb7bc1e6b9858551880684a0400ab0c) refactor(sync): automatic sync before transfer/retry/reattach/promote ([#365](https://github.com/iotaledger/wallet.rs/pull/365)) on 2021-03-02
-- Added a `TransferProgress` event type, triggered on transfer steps progress.
+- Added a `TransactionProgress` event type, triggered on transfer steps progress.
```
Comment (thibault-martinez) [inline bindings/nodejs/CHANGELOG.md]: Maybe don't change past CHANGELOGs

File: bindings/python/README.md [hunk 6]
```diff
@@ -119,17 +119,17 @@ Deletes an account.
 
 Returns the [AccountsSynchronizer](#accountssynchronizer) to setup the process to synchronize the accounts with the Tangle.
 
-### internal_transfer(from_account_id, to_account_id, amount): WalletBlock
+### internal_transaction(from_account_id, to_account_id, amount): WalletBlock
 
-Transfers an amount from an account to another.
+Transactions an amount from an account to another.
```
Comment (thibault-martinez) [inline bindings/python/README.md]: Wrong

File: bindings/nodejs/CHANGELOG.md [hunk 7]
```diff
@@ -36,7 +36,7 @@
   - [3db454e2](https://github.com/iotaledger/wallet.rs/commit/3db454e26131d5f8706e0b4ee0f4390a77416229) add GetBalance to API timeouts ([#667](https://github.com/iotaledger/wallet.rs/pull/667)) on 2021-07-19
 - Added `mqttDisabled` option to disable mqtt.
   - [349f8307](https://github.com/iotaledger/wallet.rs/commit/349f83074a378ca228dd86c3c975411de9b184fe) Add option to disable mqtt ([#665](https://github.com/iotaledger/wallet.rs/pull/665)) on 2021-06-22
-- Add optional OutputKind for transfers to enable the creation of dust allowance outputs.
+- Add optional OutputKind for transactions to enable the creation of dust allowance outputs.
```
Comment (thibault-martinez) [inline bindings/nodejs/CHANGELOG.md]: Don't change

File: bindings/python/README.md [hunk 8]
```diff
@@ -119,17 +119,17 @@ Deletes an account.
...
+### internal_transaction(from_account_id, to_account_id, amount): WalletBlock
 
 Transfers an amount from an account to another.
 
 | Param           | Type  | Default     | Description                                      |
 | --------------- | ----- | ----------- | ------------------------------------------------ |
-| from_account_id | `str` | `undefined` | The source of account id in the transfering      |
-| to_account_id   | `str` | `undefined` | The destination of account id in the transfering |
-| amount          | `int` | `undefined` | The transfer amount                              |
+| from_account_id | `str` | `undefined` | The source of account id in the transactioning      |
```
Comment (thibault-martinez) [inline bindings/python/README.md]: transactioning 🤔

File: documentation/docs/libraries/python/api_reference.md [hunk 9]
```diff
@@ -446,7 +446,7 @@ Transfer tokens.
 
 | Param        | Type                    | Default     | Description                  |
 | ------------ | ----------------------- | ----------- | ---------------------------- |
-| transfer_obj | `[Transfer](#transfer)` | `undefined` | The transfer we want to make |
+| transfer_obj | `[Transaction](#transfer)` | `undefined` | The transfer we want to make |
```
Comment (thibault-martinez) [inline documentation/docs/libraries/python/api_reference.md]: Also change the link

File: src/account/operations/transaction/mod.rs [hunk 10]
```diff
@@ -50,16 +48,16 @@ impl AccountHandle {
     /// Send a transaction, if sending a block fails, the function will return None for the block_id, but the wallet
     /// will retry sending the transaction during syncing.
     /// ```ignore
-    /// let outputs = vec![TransferOutput {
-    ///     address: "atoi1qpszqzadsym6wpppd6z037dvlejmjuke7s24hm95s9fg9vpua7vluehe53e".to_string(),
+    /// let outputs = vec![TransactionOutput {
+    ///     address: "toi1qpszqzadsym6wpppd6z037dvlejmjuke7s24hm95s9fg9vpua7vluehe53e".to_string(),
```
Comment (thibault-martinez) [inline src/account/operations/transaction/mod.rs]: Missing 'a' ?

**Review by Thoralf-M — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**